### DISCUSSION TITLE
What's the correct way to return an edge type in a mutation?

### DISCUSSION THREAD

**Discussion by DennisMartinez** [rmosolgo/graphql-ruby#4595] [Q&A] [answered]
Right now I've manually constructed the type to look like an edge, but I was wondering if there is an easier way where I don't have to manually construct the node or cursor?

Here's a simplified version of what I currently have.
```
module Mutations
  class CreateUser < BaseMutation
    field :user, Types::UserType.edge_type

    def resolve
      user = User.create!
  
      {
        user: {
          cursor: '123',
          node: user
        }
      }
    end
  end
end
```

**Comment by rmosolgo** [comment, marked as answer]
👋  GraphQL-Ruby includes `GraphQL::Relay::RangeAdd` to help with this: https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/relay/range_add.rb

I hope that helps!

**Reply by DennisMartinez** [reply]
Thank you @rmosolgo! This solved my problem.
